src/pages/RestaurantsPage/RestaurantsPage.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Fade } from "react-awesome-reveal";
+
+import "./RestaurantsPage.scss";
+import { RootState, AppDispatch } from "../../redux-toolkit/store";
+import { fetchRestaurantsPageData } from "../../redux-toolkit/thunks/restaurantsPageThunk";
+import { fetchHomePageData } from "../../redux-toolkit/thunks/homePageThunk";
+import Card from "../../components/Card/Card";
+import { CardProps } from "../../models/types";
+
+const RestaurantsPage = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const [activeFilter, setActiveFilter] = useState("All");
+
+  const { restaurants } = useSelector((state: RootState) => state.restaurantsPage);
+  const { popularRestaurants } = useSelector((state: RootState) => state.homePage);
+
+  useEffect(() => {
+    dispatch(fetchRestaurantsPageData());
+    dispatch(fetchHomePageData());
+    window.scrollTo(0, 0);
+  }, [dispatch]);
+
+  const handleFilterClick = (filter: string) => {
+    setActiveFilter(filter);
+  };
+
+  const filteredCards = () => {
+    if (activeFilter === "New") {
+      return restaurants.cards.slice(-9).map((card: CardProps) => <Card {...card} className='restaurants-page-card' key={card.title} />);
+    } else if (activeFilter === "Most Popular") {
+      return popularRestaurants.cards.map((card: CardProps) => <Card {...card} className='restaurants-page-card' key={card.title} />);
```

Comment:
If popular is if the restaurant is return with poplar from the be, so u only needs to filter where isPopular